### ISSUE TITLE
VkFormat not an instance of Num

### ISSUE THREAD

**Issue by Rotaerk** [achirkin/vulkan#43]
What's the reason for removing this instance?  I can work around it by explicitly wrapping/unwrapping a number with VkFormat, but it would be convenient if VkFormat itself were a number.

**Comment by achirkin** [comment]
A while ago I made some efforts to reduce the size and compile time of the generated code and documentation #37. As a part of that I removed `Num` instances from all Enums and Bitmasks. Here is the logic for not using `Num`:
  * `fromInteger`/`fromIntegral` are not safe, because they allow you to create a non-existing enum; you can always use pattern synonyms instead.
  *  For bitmasks/bits you can now use `bitToMask` and `maskToBits` functions, also notice `VK_ZERO_FLAGS` pattern for bitmasks.
  *  Other `Num` operations are misleading; if you want to iterate through possible enum values, use `Enum` instances.
  * Thus, the cases when you really need a `Num` instance must be rare; then you can unpack the newtype constructors or use a `Coercible` instance.

Just out of curiosity: what is your use case for `Num VkFormat`? I could not find it in your repo via git search :)

**Comment by Rotaerk** [comment]
Okay, I think your reasons for removing the Num instance vastly outweigh the inconvenience I'm thinking of.  My use-case is probably pretty unusual, I guess.  I made a library for reading KTX files, a texture format by Khronos that has GL constants built in to it.  My library handles conversion from these GL constants to the appropriate VK constants.  However, I didn't want to tie my library to any GL or VK bindings, so I redefined all the relevant GL and VK constants within the library.  The type of these constants is `(Num n, Show n, Eq n) => n` so that they can be converted to the appropriate data type for whichever binding is being used.  https://github.com/Rotaerk/ktx-rw/blob/master/ktx-rw/src/Codec/Image/Ktx/VkConstants.hs

So anyway, I think this can be closed.  I can work around this by explicitly wrapping/unwrapping with the VkFormat constructor.  The benefit isn't worth the cost.
